lib/evaluate/intrinsics.cc

```diff
@@ -772,9 +772,9 @@ static const SpecificIntrinsicInterface specificIntrinsicFunction[]{
     {{"idnint", {{"a", DoublePrecision}}, DefaultInt}, "nint"},
     {{"ifix", {{"a", DefaultReal}}, DefaultInt}, "int", true},
     {{"index", {{"string", DefaultChar}, {"substring", DefaultChar}},
-        DefaultInt}},
+        SubscriptInt}},
     {{"isign", {{"a", DefaultInt}, {"b", DefaultInt}}, DefaultInt}, "sign"},
-    {{"len", {{"string", DefaultChar, Rank::anyOrAssumedRank}}, DefaultInt,
+    {{"len", {{"string", DefaultChar, Rank::anyOrAssumedRank}}, SubscriptInt,
```

Comment:
Sorry for commenting on a merged PR, but isn't it a deviation from the standard ? 16.9.112 4 :

> Result Characteristics. Integer scalar. If KIND is present, the kind type parameter is that specified by the
> value of KIND; otherwise the kind type parameter is that of default integer type.

(Same for INDEX). I am not saying it is wrong, but maybe a note should be added about the concept of subscriptIntegerKind into Extension.md ? Something like:

> The integer result of LEN, INDEX and LOC intrinsics when KIND is not present has kind 8 (therefore not necessarily the default integer type) . This is to make address calculations 64-bit safe.

Reply:
It is an intentional deviation, and not the first for this issue.  Fortran's default INTEGER kind is simply broken as a choice for anything related to sizes (including loop controls).  But Fortran defines the default INTEGER as being the same size in storage as the default REAL, and that's always going to be 32 bits for probably forever.  I'll ensure that it is documented.